packages/components/src/components/hds/app-header/index.hbs

```diff
@@ -0,0 +1,16 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+<div class={{this.classNames}} ...attributes>
+  <div class="hds-app-header__sub-section">
+    {{yield to="globalBefore"}} {{! Intended for: ProductMenu }}
```

Comment:
My original intention was to enforce usage of the HomeLink/Logo and a recommended order for the content. In thinking it over though, you are right that with the current 3 slots, the order isn't actually enforced. I also talked it over with @jorytindall who brought up that there's less likely potential for misuse of this type of component than with more general use components (since consumers would add it just in one or a few places and then mostly leave it alone).

I'll go ahead and refactor it into one general slot for the global items.